packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/ios/enrollmentintegrationsteps.md

```diff
@@ -0,0 +1,523 @@
+
+The code in this guide manages the sign-on flow using a singleton class (`SignInController`). The class handles asynchronous responses from the SDK by implementing the delegate functions of the `IDXClientDelegate` protocol. The SDK also provides completion handlers and Swift Concurrency (`async/await`).
+
+### 1: Display the initial sign-in UI
+
+Enable the user to start the sign-in flow, such as presenting a Sign In button or a form with fields for a username and password (you won't use those values until step 3 when you [authenticate the user credentials](#_3-authenticate-the-user-credentials)).
+
+### 2: Start the sign-on flow
+
+Initialize the SDK client and request the initial server response after the user starts the sign-in process.
+
+This code also sets the caller as the delegate for asynchronous callbacks from the SDK:
+
+```swift
+// Initialize the SDK client using a Configuration object loaded previously.
+IDXClient.start(with: configuration) { (client, error) in
+    guard let client = client else {
+		self.finish(with: error)
+		return
+	}
+
+	self.client = client
+
+    // Set this object as the delegate for asynchronous calls.
+	client.delegate = self
+
+	// Request the first response from the Okta org.
+	client.resume()
+}
+```
+
+The `configuration` (`IDXClient.IDXClientConfiguration`) parameter in the call to `IDX.start(with:)` contains the configuration information for your Okta org, and is usually initialized from an `Okta.plist` file. The call to `client.resume()` requests the first response of the sign-sequence.
+
+### 3: Authenticate the user credentials
+
+The SDK uses a `Response` object to represent the server request for the current sign-on step. It contains a collection of remediations (`Remedation`) that specify the type of step, any general messages, and a nested collection of forms (`Remedation.Form`) containing fields (`Remedation.Form.Field`) that represent the various elements of the UI, such as an entry field with a label. Completing a step usually requires setting the value of one or more fields. The remediation is then sent back to the server with the updated values by calling `proceed()`. The server processes the values and then sends the next step.
+
+The kind of sign-on step is represented by the `type` property of the remediation. In this step the server is requesting the username and password, an `.identify` remediation type.
+
+This code shows the singleton delegate function that handles the an SDK response indicating a next step and includes all the types of remediations that are part of a Google Authenticator flow:
```

Comment:
Nice catch of a crash paragraph. How about:

> The SDK calls the `idx(client: response:)` delegate function with the server response that specifies the next step. This code uses a `switch` statement to handle the different types of remediation that are part of a Google Authenticator enrollment flow:

Reply:
TOns better. Nice one :-)

Reply:
You haven't actually updated this text yet.

Reply:
I didn't update because I was waiting for a "this looks good" ;-).

I also slightly modified the text which will show up shortly

> The SDK calls the `idx(client: response:)` delegate function with the server response that specifies the next step. This code first checks for a successful sign-on and for other conditions, and then uses a `switch` statement to handle the different types of remediation that are part of a Google Authenticator enrollment flow: